src/utils/i18n-utils.js

```diff
@@ -25,16 +25,34 @@ export function localeIsValid(localeCode) {
 
 /**
  * Updates i18n global var and html lang
- * @param {{ params: { locale: String } }} to - where the route navigates to
+ * @param {String} locale - locale used
  * @param {Object{}} env - context
- * @param {{ code: String, name: String }[]} locales
  */
-export function updateCurrentLocale(to, env) {
-  const currentLocale = to.params.locale || defaultLocale;
+export function updateLocale(locale = defaultLocale, env) {
```

Comment:
I'm not sure we would need to export this function since it is only used once in the same file—maybe it doesn't even need to be a separate function?